orderfilter/shared_test.go

```diff
@@ -0,0 +1,177 @@
+package orderfilter
+
+import (
+	"fmt"
+	"math/big"
+	"strings"
+	"testing"
+
+	"github.com/0xProject/0x-mesh/constants"
+	"github.com/0xProject/0x-mesh/zeroex"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/math"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// This uses the same test cases as TestFilterValidateOrder, but it marshals and
+// then unmarshals the orderfilter before performing validation. This provides a
+// sanity check that order filters work properly after being encoded and then
+// decoded. More rigorous testing of these properties is tested in integration
+// tests of ordersync.
+func TestMarshalAndUnmarshalFilter(t *testing.T) {
```

Comment:
@albrow I am planning to de-dupe this logic with that in the other test. I didn't want this to block the review, so I will say that the only change that was made is that we now encode and then decode the orderfilter into a new object to ensure that the decoded filter works properly (passed the same tests as a filter that has not been encoded).